xmlpoke/xmlpoke.d.ts

```diff
@@ -0,0 +1,45 @@
+// Type definitions for xmlpoke 0.1.12
+// Project: https://github.com/mikeobrien/node-xmlpoke
+// Definitions by: Garth Kidd <https://github.com/garthk/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="../xpath/xpath.d.ts" />
+
+declare module XmlPoke { // ghost module
```

Comment:
I regret I'm not an expert; I'm just trying to follow the published [best practices](http://definitelytyped.org/guides/best-practices.html) (emphasis mine):

> Also called non-instantiated modules. Instead of polluting the global namespace with many interfaces, **it is okay to create a module that contains interfaces only**. This does not introduce a variable on the global namespace (see safety in below sample) and this module can only be used for types.

I've confirmed, as the safety sample shows, that I haven't introduced a variable on the global namespace: 

```
var n = XmlPoke; // [ts] Cannot find name 'XmlPoke'.
```

@basarat, you wrote the paragraph on ghost modules. Is there some nuance I'm missing? I created a module with [mostly] only interfaces… are types permitted? @sandersn is concerned I'm "polluting the global namespace". I'm not sure I am. 

Because `xmlpoke` exports a function, I can't put the interfaces and types in the module _and_ export them. In turn, consumers of the typings wouldn't be able to use them, e.g. in:

```ts
const map: PathToValueMap = {};
// programmatically build up map, with typings support for its values
xmlpoke(text, xml => xml.set(map));
```

I believe this kind of typing support is good form.

If it comes up, the `namespace xmlpoke {}` is a workaround for Microsoft/TypeScript#5073.